crates/wash/src/lib/start/nats.rs

```diff
@@ -378,7 +398,14 @@ fn nats_url(os: &str, arch: &str, version: &str) -> String {
         "x86_64" => "amd64",
         _ => arch,
     };
-    format!("{NATS_GITHUB_RELEASE_URL}/{version}/nats-server-{version}-{os}-{arch}.tar.gz")
+    if let Some(mirror) = mirror {
+        format!(
+            "{}/{NATS_GITHUB_RELEASE_PATH}/{version}/nats-server-{version}-{os}-{arch}.tar.gz",
+            mirror.trim_end_matches('/').trim()
```

Comment:
This should be swapped so `trim` happens first. With a pattern match for the `trim_end_matches` of `/`, it would see the whitespace and then stop. You could also have the pattern matcher handle both, but I think swapping would be fine